### PR TITLE
[package-json-lint-config-terra] upgraded to package-json-lint v2

### PR DESCRIPTION
**Pull request by sdadn**
### Summary

This PR upgrades the peerDependency for `package-json-lint` in `package-json-lint-config-terra` to v2 to resolve a conflicting peer-dependency issue.

CleanShot 2024-01-31 at 14 14 16

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by kenk2 — APPROVED** [review]

**Comment by kenk2** [comment]
Approved, but do we need an upgrade guide for this?

File: packages/package-json-lint-config-terra/package.json [hunk 1]
```diff
@@ -40,9 +40,8 @@
   },
   "devDependencies": {
     "@cerner/eslint-config-terra": "^5.6.0",
```
Comment (kenk2) [inline packages/package-json-lint-config-terra/package.json]: tiny nit: I think CI is panicking over this dangling comma

**Comment by sdadn** [comment]
> Approved, but do we need an upgrade guide for this?
I originally thought we wouldn't need one but now I think about it, it wouldn't hurt to have one for clarity. Good catch!

Added here: 4479973

File: packages/package-json-lint-config-terra/package.json (hunk 1 above)
Comment (sdadn) [reply packages/package-json-lint-config-terra/package.json]: fixed here: f8126c01d29ba060e185aeb02cdb1af6d215f9c1

**Review by nikhitasharma — APPROVED** [review]

**Pull request merged**